lib/common/streams/chunkstream.js

```diff
@@ -115,6 +115,11 @@ ChunkStream.prototype.error = function () {
 
 ChunkStream.prototype.destroy = function () {
   this.writable = this.readable = false;
+
+  if (this._allocator && this._allocator.destroy) {
```

Comment:
It is safer to check whether `this._allocator.destroy` is a function, since anyone can add a string or bool value with this name.